packages/babylon/src/parser/lval.js

```diff
@@ -150,7 +150,8 @@ export default class LValParser extends NodeUtils {
         if (
           arg.type !== "Identifier" &&
           arg.type !== "MemberExpression" &&
-          arg.type !== "ArrayPattern"
+          arg.type !== "ArrayPattern" &&
+          arg.type !== "ObjectPattern"
```

Comment:
We could probably transform that into a Array lookup for readability, but lgtm. Thanks for your contrib @ksashikumar